src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -65,7 +65,7 @@ static IDispatcher GetDispatcher(IServiceProvider services)
 				services.CreateLogger<Dispatcher>()?.LogWarning("Replaced an existing DispatcherProvider with one from the service provider.");
 			}
 
-			return Dispatcher.GetForCurrentThread()!;
+			return Dispatcher.GetForCurrentThread() ?? services.GetRequiredService<ApplicationDispatcher>().Dispatcher;
```

Comment:
I added the coalescing here, because we don't want to break users that are retrieving the scoped IDispatcher from the root service provider on a background thread. Because we are no longer saturating the IDispatcher on the root provider we need to account for this.

Reply:
Not sure we should do this because we will then cache the app dispatcher in the scope. The correct fix is to see who is calling the dispatcher - probably some setup code, and use the app dispatcher there. I added an extension method `GetRequiredApplicationDispatcher` that shuold be used if you are trying to get the dispatcher if you have no windows.

This method is not meant to "get a dispatcher, any dispatcher" it is meant to only return the window, scoped dispatcher. It is up to the caller to make the decision to fall back to the app dispatcher.

What was the stack trace of the crash?

Reply:
Yea, I agree, the problem I ran into was figuring out `The correct fix is to see who is calling the dispatcher`. As far as I can tell it's impossible to get context from inside the factory on who is calling the dispatcher. The main scenario this is meant to fix is if someone calls `IPlatformApplication.Current.Services.GetService<IDispatcher>()` on a background thread. We added code in this PR to `FindDispatcher` to handle that path, which should cover the majority of scenarios.

The new code we added to correctly pre-initialize the window scope dispatchers should keep those paths safe from hitting this scenario `Not sure we should do this because we will then cache the app dispatcher in the scope`

For example, in currently public MAUI if you call "ScopedService.GetService<IDispatcher>()" from a background thread that will return "NULL". After the ServiceScope PR that now returns the properly scoped service. So, this scenario is already "broken" publicly and made better by the other PR.